public/app/plugins/panel/table/models.gen.ts

```diff
@@ -4,23 +4,48 @@
 //~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 import { TableCellDisplayMode, TableSortByFieldState } from '@grafana/ui';
+import { ReduceDataOptions } from '@grafana/data';
 
 // Only the latest schema version is translated to TypeScript, on the premise
 // that either the dashboard loading process, or (eventually) CUE-defined
 // migrations ensure that bulk of the frontend application only ever
 // need directly consider the most recent version of the schema.
 export const modelVersion = Object.freeze([1, 0]);
 
+export enum FooterMode {
+  auto = 'auto',
+  summary = 'summary',
+  frame = 'frame',
+  none = 'none',
+}
+
+export enum FooterLabelMode {
+  auto = 'auto',
+  first = 'first',
+  inline = 'inline',
+}
+
 export interface PanelOptions {
   frameIndex: number;
   showHeader: boolean;
   showTypeIcons?: boolean;
   sortBy?: TableSortByFieldState[];
+  footerMode?: FooterMode;
+  footerFrame?: number;
+  footerSummary?: TableFooterSummary; // TODO: should be array (options builder is limited)
+  footerLabelMode?: FooterLabelMode;
+  reduceOptions?: ReduceDataOptions;
```

Comment:
isn't the reducer part of the `footerSummary`?

Reply:
this is referenced by the stats editor that @dprokop suggested

Reply:
updated.  just needs to be ReduceDataOptions type when calling "getFieldDisplayValues":

```
    return getFieldDisplayValues({
      fieldConfig,
      reduceOptions: options.footerSummary?.reduceOptions || { calcs: [] },
      replaceVariables,
      theme: config.theme2,
      data: [data.series[currentIndex]],
      timeZone,
    });
```